<?xml version="1.0"?>

<launch>
	<arg name="x" default="0.0"/>
	<arg name="y" default="0.0"/>
	<arg name="z" default="1"/>
	<arg name="name" default="sim_cam"/>
	<arg name="model" default="$(find uav_id)/urdf/simulated_camera.gazebo.xacro"/>

	<arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
	<arg name="world_frame" default="world"/> <!-- This should actually be "/world". See https://github.com/ros-simulation/gazebo_ros_pkgs/pull/324 -->
	<arg name="base_link_frame" default="$(arg tf_prefix)/base_link"/>

	<!-- send the robot XML to param server -->
	<param name="robot_description" command="$(find xacro)/xacro '$(arg model)' base_link_frame:=$(arg base_link_frame) world_frame:=$(arg world_frame)" />

	<!-- push robot_description to factory and spawn robot in gazebo -->
	<node name="spawn_robot" pkg="gazebo_ros" type="spawn_model"
		args="-param robot_description
			-urdf
			-x $(arg x)
			-y $(arg y)
			-z $(arg z)
			-model $(arg name)"
		respawn="false" output="screen"/>

</launch>

